MediaManager/Platforms/Android/MediaManagerImplementation.cs

```diff
@@ -164,7 +165,7 @@ public override Task Pause()
 
         public override Task Play()
         {
-            if(!this.IsPlaying())
+            if (!this.IsPlaying() && this.State != MediaPlayerState.Paused)
```

Comment:
I think there might be something wrong with the if statement. Should it not be `if (this.IsStopped())` then it should call prepare. Otherwise just Play?

Reply:
Hmm i dont know what `IsStopped()` look at is it only `Stopped `State, then you will need to consider other state, like failed etc. 
then you should `prepare `eg start new Session.

But you know better.

Reply:
Maybe it needs to be a combination of things. Maybe add a new extension method for this which does multiple checks?

Reply:
Yah, But if the user Press Play() no matter what state other then paused. it should still `prepare` a new session.

So when you say combination of things, what is it you are after here. i cant think of somthing. 

but a new global extension, that will work on all platfroms and check for state is still a good thing. 

But as i im unable to test the other platforms , i cant do a big changes.